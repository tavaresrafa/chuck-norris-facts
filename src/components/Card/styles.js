import styled from 'styled-components';
import { colors } from '../../styles/colors';

export const Card = styled.blockquote`
  text-align: justify;
  background: ${colors.gray1};
  border-left: 1rem solid ${colors.orange};
  padding: 0.5em 1rem;
  font-size: 1.6rem;
  margin-bottom: 1.6rem;
  display: flex;
  align-items: center;
  align-self: flex-start;
  transition: 0.2s ease;
  max-width: 100%;
  box-shadow: 1px 1px 3px 0 rgba(0, 0, 0, 0.2);

  :hover {
    color: ${colors.orange};
  }

  p {
    word-wrap: anywhere;
    width: 100%;
  }
`;
