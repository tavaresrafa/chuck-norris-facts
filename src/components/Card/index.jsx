import React from 'react';
import * as S from './styles';

const Card = ({ text }) => (
  <S.Card>
    <p data-testid="card">{text}</p>
  </S.Card>
)

export default Card;
