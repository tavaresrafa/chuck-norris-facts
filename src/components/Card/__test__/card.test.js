import React from 'react';
import ReactDOM from 'react-dom';
import Card from '../../Card';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Card />, div);
});

it('renders card correctly', () => {
    const { getByTestId } = render(<Card text="Chuck Norry Jokes" />);
    expect(getByTestId('card')).toHaveTextContent('Chuck Norry Jokes');
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<Card text="Chuck Norry Jokes" />).toJSON();
    expect(tree).toMatchSnapshot();
});