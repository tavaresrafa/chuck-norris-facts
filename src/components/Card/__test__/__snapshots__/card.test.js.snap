// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<blockquote
  className="sc-AxjAm hvldEi"
>
  <p
    data-testid="card"
  >
    Chuck Norry Jokes
  </p>
</blockquote>
`;
